src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.File.cs

```diff
@@ -46,5 +57,40 @@ public void Delete()
             foreach (var file in filesToRemove)
                 File.Delete(file);
         }
+
+        [GlobalSetup(Targets = new[] { nameof(WriteAllBytes), "WriteAllBytesAsync" })]
+        public void SetupWriteAllBytes()
+        {
+            _testFilePath = FileUtils.GetTestFilePath();
+            _userBuffers = new Dictionary<int, byte[]>()
+            {
+                { HalfKibibyte, ValuesGenerator.Array<byte>(HalfKibibyte) },
+                { FourKibibytes, ValuesGenerator.Array<byte>(FourKibibytes) },
+                { SixteenKibibytes, ValuesGenerator.Array<byte>(SixteenKibibytes) },
+                { OneMibibyte, ValuesGenerator.Array<byte>(OneMibibyte) },
+                { HundredMibibytes, ValuesGenerator.Array<byte>(HundredMibibytes) },
+            };
+        }
+
+        [Benchmark]
+        [Arguments(HalfKibibyte)]
+        [Arguments(FourKibibytes)]
+        [Arguments(SixteenKibibytes)]
+        [Arguments(OneMibibyte)]
+        [Arguments(HundredMibibytes)]
+        public void WriteAllBytes(int userBuffer) => File.WriteAllBytes(_testFilePath, _userBuffers[userBuffer]);
```

Comment:
nit: I would just  rename the arg name to `size` as it's part of benchmark ID

```suggestion
        public void WriteAllBytes(int size) => File.WriteAllBytes(_testFilePath, _userBuffers[size]);
```